### PR TITLE
changes to the Hindley-Milner Inference chapter

### PR DESCRIPTION
**Pull request by chsievers**
Small and not so small changes to the beginning of chapter 7.

### CODE REVIEW

**Comment by sdiehl** [comment]
Sweet, I'll look over the changes this weekend. Have you tried rendering the latex either using pdflatex or mathjax.

Sometimes line like this:

```
we get
$$[s]\alpha=\alpha\rightarrow\beta$$
and
```

Need an extra space between the display equations to render in the browser.

```
we get

$$[s]\alpha=\alpha\rightarrow\beta$$

and
```

**Comment by chsievers** [comment]
At least the mathjax version renders fine for me.

File: 006_hindley_milner.md [hunk 1]
```diff
@@ -437,20 +436,33 @@ $$
...
-long function type.
-
-$$
-[x/x \rightarrow x]
-$$
-
-The occurs check will forbid the existence of many of the pathological livering
-terms we discussed when covering the untyped lambda calculus, including the
-omega combinator.
+If we want to unify a type variable $\alpha$ with a type $\tau$, we usually
```
Comment (sdiehl) [inline 006_hindley_milner.md]: Much more clear.

File: 006_hindley_milner.md [hunk 2]
```diff
@@ -437,20 +436,33 @@ $$
...
+$$[s]\alpha\rightarrow\beta=(\alpha\rightarrow\beta)\rightarrow\beta.$$
+
+Indeed, whatever substitution $s$ we try, $[s]\alpha\rightarrow\beta$ will
+always be longer than $[s]\alpha$, so no unifier exists.
+The only chance would be to substitute with an infinite type:
+$[\alpha/(\dots((\alpha\rightarrow\beta)\rightarrow\beta)\rightarrow\dots
+ \rightarrow\beta)\rightarrow\beta]$
+would be a unifier, but our language has no such types.
+
+If the unification fails because of the occurs check, we say that unification
```
Comment (sdiehl) [inline 006_hindley_milner.md]: Yes, better to clarify that point before they see the InfiniteType error.

**Pull request merged**